charts/controller/templates/controller-deployment.yaml

```
@@ -1,4 +1,4 @@
-apiVersion: extensions/v1beta1
+apiVersion: apps/v1
```

I think this should be something like this to keep backwars compatibility with older clusters:

```suggestion
{{- if .Capabilities.APIVersions.Has "apps/v1" }}
apiVersion: apps/v1
{{- else }}
apiVersion: extensions/v1beta1
{{- end }}
```

---

Going to try the above and see if it works for us. :+1: Sometime before we have had problems using `. Capabilities.APIVersions.Has`.

---

The docs mention that checking for resources (`apps/v1/Deployment`) requires at least Helm v2.15, but just using `apps/v1` (without a resource) should be fine, if that's specific enough.

---

You can perhaps just check for "v1" – I think you might not find specific API groups, but you can count on support for `apps/v1` if APIVersions has "v1"

There was an issue I found on the Prometheus helm chart repo that covered this, I will look to see what they settled on to solve the backwards compatibility question, and if it differs

---

Following this suggestion in 5930851c . :+1:  I found this similar example in a Datadog agent chart that looks very similar to exactly what we are doing here:

https://github.com/helm/charts/blob/2215d0f752370ae75fa54a9f62b057ed14f8e744/stable/datadog/templates/agent-clusterchecks-deployment.yaml#L2-L9

I think they are adding the last `else` in the end to make sure it tries to get deployed using the default apiVersion as `apps/v1` even if the deployer agent (Helm or whatever...) does not have permission to view `.Capabilities.APIVersions`?